dialyzer/src/main.ts

```diff
@@ -92,15 +92,26 @@ async function run(): Promise<void> {
   } = await elixirVersions();
   const mixLockHash = await hashFiles(["mix.lock", "apps/*/mix.lock"]);
   const dialyzerPaths = [" _build/*/*.plt*"];
-  const cacheKey = `${architecture}-dialyzer-${otp_release}-${erts_version}-${elixir_version}-${mixLockHash}`;
-  const restoreKeys = [
-    `${architecture}-dialyzer-${otp_release}-${erts_version}-${elixir_version}-`,
-    `${architecture}-dialyzer-${otp_release}-${erts_version}-`,
-    // previous version of the Dialyzer cache
-    `${architecture}-dialyzer-${otp_release}-${elixir_version}-`,
-    `${architecture}-dialyzer-${otp_release}-`,
-    `${architecture}-dialyzer-`,
-  ];
+  const cacheKeyVersion = core.getInput("cache-key-version");
+  let cacheKey = `${architecture}-dialyzer-${otp_release}-${erts_version}-${elixir_version}-${mixLockHash}`;
+  if (cacheKeyVersion !== "") {
+    cacheKey = `${cacheKey}-${cacheKeyVersion}`;
```

Comment:
question: do you want `cacheKeyVersion` first instead? otherwise, changing it will still fallback to a previous cache.

Reply:
Good point, I missed thinking about the behavior if cache-key-version was used without disabling fallback